src/app/api/submitPayment/route.js

```diff
@@ -126,6 +127,123 @@ async function maxHopeFuelID() {
   return result[0]["maxHopeFuelID"];
 }
 
+// get currency by wallet ID
+async function getCurrencyByWalletId(walletId) {
+  console.log("Wallet ID: ", walletId);
+  const query = `
+    SELECT 
+      C.CurrencyId, C.CurrencyCode
+    FROM 
+      Wallet AS W
+    JOIN 
+      Currency AS C 
+    ON
+      W.CurrencyId = C.CurrencyId
+    WHERE
+      W.WalletId = ?;
+  `;
+
+  const values = [walletId];
+
+  try {
+    const result = await db(query, values);
+    return result.length > 0 ? result[0] : null;
```

Comment:
Instead of null, if there is no result returning from DB, I encourage to return ed undefined (return result[0] and handle error in api request by responding with status code and currency not found message). I hope it would be obvious to debug when we need to find error. 
```
const currency = await getCurrencyByWalletId(id);
if (!currency) {
  // handle not found
}
```